extensions/sql-database-projects/src/models/project.ts

```diff
@@ -130,23 +135,51 @@ export class Project implements ISqlProject {
 			console.error(utils.getErrorMessage(e));
 		}
 
-		// glob style getting files and folders for new msbuild sdk style projects
-		if (this._isMsbuildSdkStyleProject) {
-			const files = await utils.getSqlFilesInFolder(this.projectFolderPath, true);
-			files.forEach(f => {
-				this._files.push(this.createFileProjectEntry(utils.trimUri(Uri.file(this.projectFilePath), Uri.file(f)), EntryType.File));
-			});
+		// find all import statements to include
+		try {
+			const importElements = this.projFileXmlDoc!.documentElement.getElementsByTagName(constants.Import);
+			for (let i = 0; i < importElements.length; i++) {
+				const importTarget = importElements[i];
+				this._importedTargets.push(importTarget.getAttribute(constants.Project)!);
+			}
```

Comment:
I'm thinking that this should be shifted to a helper function, and `this._importedTargets` shouldn't be updated until everything's been read.  I'm not a fan of throwing an error while reading, but leaving the project field partially updated.

Reply:
so if there's an error while reading the imported targets, it should be set to [] instead of whatever was already read up to that point?